examples/seismic/viscoacoustic/wavesolver.py

```diff
@@ -121,3 +132,62 @@ def forward(self, src=None, rec=None, v=None, r=None, p=None, qp=None, b=None,
                                               b=b, vp=vp,
                                               dt=kwargs.pop('dt', self.dt), **kwargs)
         return rec, p, summary
+
+    def adjoint(self, rec, srca=None, u=None, q=None, vp=None, qp=None, b=None, r=None,
+                **kwargs):
+        """
+        Adjoint modelling function that creates the necessary
+        data objects for running an adjoint modelling operator.
+
+        Parameters
+        ----------
+        rec : SparseTimeFunction or array-like
+            The receiver data. Please note that
+            these act as the source term in the adjoint run.
+        srca : SparseTimeFunction or array-like
+            The resulting data for the interpolated at the
+            original source location.
+        v: TimeFunction, optional
+            The computed wavefield.
+        vp : Function or float, optional
+            The time-constant velocity.
```

Comment:
qp, r, b, q missing from docstring.

What's `q` again, pressure? If pressure would not use `q` usually used for source or attenuation and name it something like `pa`. Would also not name particle velocity `u`, that's confusing with most forward using `u` for forward wavefield, maybe `va`?

Reply:
Okay, we will follow your suggestions.